### ISSUE TITLE
Internet Identity does not work on prod with this.

### ISSUE THREAD

**Issue by wyattbenno777** [MioQuispe/create-ic-app#17]
Hello!

agentOptions: the identity is never passed to this. So on local all may seem well, but when you push to the IC all msg.caller fields will be anonymous making all msg.callers the same Principal. 

There needs to be a way to pass the currently logged in user to agentOptions as identity. This will allow the calls to be made as that person for shared(msg) in Motoko.

Let me know if there is already a way to pass the identity into the createActor.

**Comment by wyattbenno777** [comment]
Work-around is to manually create the actor for your canisters.

Ideally, this would work by default in the examples with nil as the identity to start and the signed in user 
after.

`
import { createActor as counterCA } from "canisters/counter"
const counter = counterCA(process.env.COUNTER_CANISTER_ID, {identity: 'what you get from signin'})
`

**Comment by MioQuispe** [comment]
Yep. thanks for raising the issue. I've been pondering before how to do this nicely without getting into framework specifics (using context, etc.) but seems theres no good way.

I might introduce that in a future update however.

**Comment by wyattbenno777** [comment]
Maybe just a note in the docs for now?
I bet a lot of people are assuming this works, because it will work on local then when they push to prod, they are surprised.
